scripts/generate/templates/DEFAULT/react.template.js

```diff
@@ -0,0 +1,47 @@
+import React, {useMemo} from 'react';
+import {
+  Button,
+  Text,
+  Stack,
+  extend,
+  render,
+  useLocale
+} from '@shopify/argo-admin-react';
+
+const translations = {
+  de: {
+    hello: 'Guten Tag',
+    clickMe: 'Klick mich',
+  },
+  en: {
+    hello: 'Hello',
+    clickMe: 'Click Me',
+  },
+  fr: {
+    hello: 'Bonjour',
+    clickMe: 'clique moi',
+  },
+};
+
+function Extension() {
+  const locale = useLocale();
+
+  const localizedStrings = useMemo(() => {
+    return translations[locale] || translations.en;
+  }, [locale]);
```

Comment:
Is this pattern used elsewhere? Would there be value in introducing a custom hook?

```js
function useTranslate(translationsForAllLocales) {
  const locale = useLocale();
  const translations = useMemo(() => translationsForAllLocales[locale] || translationsForAllLocales.en), [locale]);
  return (key) => translations[key];
}

function Extension() {
  const translate = useTranslate(translations)

  return (
    <>
      <Stack vertical>
        <Text>{translate('hello')}</Text>
        <Button title={translate('clickMe')} primary onPress={() => console.log('Button clicked!')}/>
      </Stack>
    </>
  )
}
```

Wrapping the translations in a function is an optional optimization, but I like it so I thought I suggest it.

Reply:
> Is this pattern used elsewhere? Would there be value in introducing a custom hook?

This is used in the subscriptions extension template. This is in place to demonstrate the usage of the useLocale hook. Developers would likely introduce their own translation library or functionality. 

That said I don't mind making the change since it's easier to read (for proficient react devs at least).

Reply:
Up to you! 😄 I'm just always looking for ways to reduce component complexity and have the component focus as much as possible on rendering and extract state management and side effects as much as I can.

Reply:
That's why you're a staff dev 😊